pandas/core/reshape/reshape.py

```diff
@@ -1046,9 +1046,7 @@ def get_empty_frame(data) -> DataFrame:
         fill_value: bool | float | int
         if is_integer_dtype(dtype):
             fill_value = 0
-        # error: Non-overlapping equality check (left operand type: "dtype[Any]", right
-        # operand type: "Type[bool]")
-        elif dtype == bool:  # type: ignore[comparison-overlap]
```

Comment:
looks like a false positive (maybe from numpy types)

Reply:
`np.bool_` does not inherit from `bool`: https://numpy.org/doc/stable/reference/arrays.scalars.html

Reply:
The error message is `Non-overlapping equality check...`. The numpy types should allow a check with a type. either `bool`, `np.bool_` and `str`, `object` or any other type. Doing the equality check does not raise a type error, but should always return bool.

```
import numpy as np

dtype = np.dtype("bool")
print(dtype)
print(dtype == bool)
print(dtype == np.bool_)
print(dtype == "bool")
print(dtype == object)
```
gives
```
bool
True
True
True
False
```

In the numpy type annotations for 1.21.3,  dtype.\_\_eq__ is not defined, but looks like this is fixed on master. https://github.com/numpy/numpy/pull/20184

https://github.com/numpy/numpy/blob/8eb6555711888f0ab5594a2bd82d230ba33a3d7c/numpy/__init__.pyi#L876-L880

Reply:
so what's the suggested action here?

Reply:
what you've done is fine. but we could also leave these ignores for now. 

Once we are checking with the latest numpy types we could maybe revert all the changes that have been made to-date to get round this. wdyt?

Reply:
sure